sdk/src/unittest/TestRunner.ls

```diff
@@ -0,0 +1,379 @@
+        public var target:Object;
+        public var name:String;
+        public var method:MethodInfo;
+        public var meta:MetaInfo;
+        
+        public var skip:Boolean;
+        
+        public var report:StatusReport = new StatusReport();
+        public var results:Vector.<AssertResult>;
+        
+        function Test() { };
+        
+        [UnitTestHideCall]
+        public function run():Object {
+            return method.invoke(target);
+        }
+        
+        public function toString():String {
+            return "[Test "+name+"]";
+        }
+    }
+    
+    public class TestResult {
+        public var scannedTypes:int;
+        public var typeTests:Vector.<TypeTest>;
+        
+        public var typeReport:StatusReport = new StatusReport();
+        public var testReport:StatusReport = new StatusReport();
+        public var assertReport:StatusReport = new StatusReport();
+    }
+    
+    public class TestRunner {
+        
+        public function TestRunner() {
+            
+        }
+        
+        
+        public static function runAll(assembly:Assembly, shuffle:Boolean = true):TestResult {
+            var result = new TestResult();
+            
+            result.scannedTypes = assembly.getTypeCount();
+            
+            IO.write("Scanning "+result.scannedTypes+" "+(result.scannedTypes == 1 ? "type" : "types")+"\n");
+            
+            result.typeTests = getTypeTests(assembly);
+            var totalTests = 0;
+            for each (var typeTest in result.typeTests) {
+                totalTests += typeTest.tests.length;
+            }
+            
+            IO.write("Found "+result.typeTests.length+" "+(result.typeTests.length == 1 ? "type" : "types"));
+            IO.write(" with a total of "+totalTests+" "+(totalTests == 1 ? "test" : "tests")+"\n");
+            
+            IO.write("\n");
+            
+            runTypes(result.typeTests, shuffle, result.typeReport, result.testReport, result.assertReport);
+            
+            reportTypes(result.typeTests, result.typeReport, result.testReport, result.assertReport);
+            
+            return result;
+        }
+        
+        [UnitTestHideCall]
+        public static function runTypes(typeTests:Vector.<TypeTest>, shuffle:Boolean = true, typeReport:StatusReport = null, testReport:StatusReport = null, assertReport:StatusReport = null) {
+            if (shuffle) typeTests.shuffle();
+            
+            var tests = new Vector.<Test>();
+            
+            var i:int;
+            var tt:TypeTest;
+            
+            if (typeReport) typeReport.total += typeTests.length;
+            
+            for (i = 0; i < typeTests.length; i++) {
+                tt = typeTests[i];
+                tt.asserts.reset();
+                
+                //IO.write((tt.skip ? "Skipping" : "Running")+" "+tt.type.getFullName()+"   "+(i+1)+" / "+typeTests.length+"\n");
```

Comment:
Different output styles, not sure which one to use, but the uncommented one is probably fine for now.